### DISCUSSION TITLE
Error Tx while waiting for EOB, timed out... 3.384 >= 3.21399. Starting new burst... RF status: O=0, U=1, L=0

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#112] [General Help]
Hi all,

I am writing to seeking your help regarding an issue I am facing while installing srsRan 5G SA. I have successfully installed srsgnb and open5gs and connected them. However, when I try to connect srsue with srsgnb.

I encounter the following problem:
It appears that srsue can connect with the phy layer, but it cannot be registered by open5gs. I have included the log information below:
![Screenshot from 2023-07-05 14-24-32](https://github.com/srsran/srsRAN_Project/assets/41448193/5fbe2751-1242-4d0c-89aa-196b0bfb8c6a)
![Screenshot from 2023-07-05 14-24-35](https://github.com/srsran/srsRAN_Project/assets/41448193/37179501-7b6f-4995-98c2-f36f2667a8d8)
![Screenshot from 2023-07-05 14-34-01](https://github.com/srsran/srsRAN_Project/assets/41448193/c16c7021-d61e-446b-ae35-471539763f5f)
![Screenshot from 2023-07-05 14-34-10](https://github.com/srsran/srsRAN_Project/assets/41448193/b7139c92-86ef-4f5c-8568-df51389401c7)
![251244861-b0d410c8-f213-4d1c-ae3f-f50eb6c6fa6e](https://github.com/srsran/srsRAN_Project/assets/41448193/41b5ba6d-23df-4fee-854f-a37e98be7200)
![251244858-53627e73-fa51-4bd3-86b0-e19562720da8](https://github.com/srsran/srsRAN_Project/assets/41448193/06906bd2-d26c-4dbb-8aec-a6fc874144eb)
![251244856-1cd0c9e1-df45-4ce9-954d-1de91c242714](https://github.com/srsran/srsRAN_Project/assets/41448193/56bafb42-a8ac-4c9d-8536-4fa0b57a0f12)
![251243654-5416e936-7286-4193-b06b-f23fc992ae2c](https://github.com/srsran/srsRAN_Project/assets/41448193/b818735f-14d8-48f4-8da6-f67fabbaeff3)
![251243653-f6a3e35f-2807-4c79-878d-66f77ee8ccdd](https://github.com/srsran/srsRAN_Project/assets/41448193/8caa836c-8fb0-479d-88e5-a44bf784ce46)
![251243649-5f0d648e-1a2e-4e27-b383-eef1e6349ae1](https://github.com/srsran/srsRAN_Project/assets/41448193/46f19fc4-4250-40d6-b635-bef3114d9255)

I sincerely appreciate your time and assistance in advance. Thank you!

**Comment by brendan-mcauliffe** [comment]
Hey @rajugbv, 

Firstly, attaching log files, pcaps, and text files instead of screenshots is usually better. Files are easier to navigate and investigate for issues. Doing this with images can be much more time-consuming and isn't the easiest to look at. 

Can you ensure you are using the newest versions of both srsUE and the srRAN Project gNB. There was a known issue that was fixed in the latest version of srsUE, this issue was preventing srsUE from connecting to the Core properly. 

Once you have upgraded to the latest versions of both, re-run your tests. If you are still seeing the same issue then attach your logs, pcaps and other info here and I can take another look.

**Reply by rajugbv** [reply]
@brendan-mcauliffe Thank you  very much for replying  i have checked both the versions of srsran Project and srsue both of them are up to date  and i have attached log files ,please check it if any other log file are need please ask it. 

**gnb test log file:-**
[enb.log](https://github.com/srsran/srsRAN_Project/files/11972703/enb.log)
**amf testlog file:-**
[amf.log](https://github.com/srsran/srsRAN_Project/files/11972721/amf.log)
**ue configuration file**
[ue.conf.log](https://github.com/srsran/srsRAN_Project/files/11972734/ue.conf.log)
**srsue testlog**
[ue.log](https://github.com/srsran/srsRAN_Project/files/11972806/ue.log)
**GNB configuration file :-** 
[gnbconf.log](https://github.com/srsran/srsRAN_Project/files/11972808/gnbconf.log)

**Comment by brendan-mcauliffe** [comment]
Hi @rajugbv, 

Can you enable the `mac` pcap for the gNB and the `mac_nr` pcap for the UE and attach them here. 

Before running the tests, can you set your CPU to performance mode on both PCs, you are seeing some lates. These are most likely not causing the issue, but it's better to try and get rid of them if possible. To set the CPU governor to performance mode use the following command: 

`echo "performance" | sudo tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor`

You should also double check that the APN in the Core is set to the same as that in the UE config. The usual issue is that the names are different, so make sure that it is called `srsapn` in both the core and UE config (or what ever name you choose). Also, make sure that both are set to use IPv4, not IPv6. 

Lastly, your version of srsRAN Project seems to be a commit from more than 2 weeks ago, the code has been updated since then, as have the configuration files. I suggest you `pull` the latest version and re-build the code. You can find a working config for the latest version of the code [here](https://github.com/srsran/srsRAN_Project/blob/main/configs/gnb_rf_b210_fdd_srsUE.yml), this is example is specifically for use with srsUE.

**Reply by rajugbv** [reply]
@brendan-mcauliffe 
i am still facing the same error i have tried with gnb_rf_b210_fdd_srsUE.yml conf file .
 i have set true to pcap in gnb and pcap for ue i have attached configuration file below please check it.
mac file is empty in gnb and i have also set  the CPU governor to performance mode 

i have update srsrran :-
**raju@raju-ThinkPad-P16s-Gen-1:~/srsran/srsRAN_Project$ git describe --tags
release_23_5-1090-g549d12131**

APN as been set to srsran and ipv4 
**Below is the screen shot in carrier details in gnb labtop**
![Screenshot from 
![Screenshot from 2023-07-07 17-00-49](https://github.com/srsran/srsRAN_Project/assets/41448193/68e31805-70e8-4060-8b68-8374ff6db455)
2023-07-07 17-00-44](https://github.com/srsran/srsRAN_Project/assets/41448193/acd345a2-d5ea-4659-a3aa-9500bf59757b)

[gnb_mac.log](https://github.com/srsran/srsRAN_Project/files/11985937/gnb_mac.log)
[gnb_ngap.log](https://github.com/srsran/srsRAN_Project/files/11985938/gnb_ngap.log)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/11985939/gnb.log)
[gnb_rf_b210_fdd_srsUE.log](https://github.com/srsran/srsRAN_Project/files/11985950/gnb_rf_b210_fdd_srsUE.log)
[ue (1).log](https://github.com/srsran/srsRAN_Project/files/11986027/ue.1.log)
[ue.conf (1).log](https://github.com/srsran/srsRAN_Project/files/11986028/ue.conf.1.log)

**Reply by rajugbv** [reply]
@brendan-mcauliffe 
HI Bredan, in the above comment  i think i have not attached files in proper format so i have attached zip files containing all file and  i have attached webui screenshot in the above comment please check and let me know if any any is needed 
thank you
[gnb_debugfiles.zip](https://github.com/srsran/srsRAN_Project/files/12016398/gnb_debugfiles.zip)
[ue_debug_files.zip](https://github.com/srsran/srsRAN_Project/files/12016424/ue_debug_files.zip)

**Comment by ismagom** [comment]
Hi @rajugbv , it seems you have some issues in the UHD driver in the UE side. Those messages "Tx while waiting for EOB.. starting new burst" should not appear. Have you tried running a UHD benchmark with the same sampling rate?

**Comment by rajugbv** [comment]
@ismagom  thanking for replaying ,for ue i am using USRP B200 mini and for gnb i am using B210

[ue_rfconf.txt](https://github.com/srsran/srsRAN_Project/files/12018535/ue_rfconf.txt)
[gnb_rf_b210_fdd_srsUE.log](https://github.com/srsran/srsRAN_Project/files/12018548/gnb_rf_b210_fdd_srsUE.log)


for tx srate i have used 11.52 and  for rx sampling i have used  11.52e6

while doing beach test i am getting Waring as "The hardware does not support the requested TX sample rate:" i have attached my log files and test log. please check it and let me know to what need to change
Thank you

raju@raju-ThinkPad-P16s-Gen-1:~/uhd/host/build/examples$ ./benchmark_rate --rx_rate 11.52e6 --tx_rate 11.52

[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[00:00:00.000216] Creating the usrp device with: ...
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz... 
[INFO] [B200] Actually got clock rate 16.000000 MHz.
Using Device: Single USRP:
  Device: B-Series Device
  Mboard 0: B200mini
  RX Channel: 0
    RX DSP: 0
    RX Dboard: A
    RX Subdev: FE-RX1
  TX Channel: 0
    TX DSP: 0
    TX Dboard: A
    TX Subdev: FE-TX1

[00:00:01.739617427] Setting device timestamp to 0...
[INFO] [B200] Asking for clock rate 46.080000 MHz... 
[INFO] [B200] Actually got clock rate 46.080000 MHz.
[WARNING] [MULTI_USRP] The hardware does not support the requested TX sample rate:
Target sample rate: 0.000012 MSps
Actual sample rate: 0.090000 MSps

[00:00:02.40028689] Testing receive rate 11.520000 Msps on 1 channels
[WARNING] [AD936X] Selected Tx sample rate (0.09 MHz) is less than
analog frontend filter bandwidth (0.2 MHz).
Setting TX spp to 2040
[00:00:02.63146101] Testing transmit rate 0.090000 Msps on 1 channels
[00:00:12.176896705] Benchmark complete.


Benchmark rate summary:
  Num received samples:     115465922
  Num dropped samples:      0
  Num overruns detected:    0
  Num transmitted samples:  956760
  Num sequence errors (Tx): 0
  Num sequence errors (Rx): 0
  Num underruns detected:   0
  Num late commands:        0
  Num timeouts (Tx):        0
  Num timeouts (Rx):        0


Done!

raju@raju-ThinkPad-P16s-Gen-1:~/uhd/host/build/examples$ ./rx_samples_to_file --freq 98e6 --rate 5e6 --gain 20 --duration 10 usrp_samples.dat

Creating the usrp device with: ...
[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[INFO] [B200] Loading firmware image: /usr/local/share/uhd/images/usrp_b200_fw.hex...
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Loading FPGA image: /usr/local/share/uhd/images/usrp_b200mini_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz... 
[INFO] [B200] Actually got clock rate 16.000000 MHz.
Using Device: Single USRP:
  Device: B-Series Device
  Mboard 0: B200mini
  RX Channel: 0
    RX DSP: 0
    RX Dboard: A
    RX Subdev: FE-RX1
  TX Channel: 0
    TX DSP: 0
    TX Dboard: A
    TX Subdev: FE-TX1

Setting RX Rate: 5.000000 Msps...
[INFO] [B200] Asking for clock rate 40.000000 MHz... 
[INFO] [B200] Actually got clock rate 40.000000 MHz.
Actual RX Rate: 5.000000 Msps...

Setting RX Freq: 98.000000 MHz...
Setting RX LO Offset: 0.000000 MHz...
Actual RX Freq: 98.000000 MHz...

Setting RX Gain: 20.000000 dB...
Error: LookupError: IndexError: multi_usrp: RX channel 18446744073709551615 out of range for configured RX frontends


raju@raju-ThinkPad-P16s-Gen-1:~$   uhd_usrp_probe
[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Loading FPGA image: /usr/local/share/uhd/images/usrp_b200mini_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz... 
[INFO] [B200] Actually got clock rate 16.000000 MHz.
  _____________________________________________________
 /
|       Device: B-Series Device
|     _____________________________________________________
|    /
|   |       Mboard: B200mini
|   |   serial: 323E38A
|   |   name: B200mini
|   |   product: 30521
|   |   revision: 3
|   |   FW Version: 8.0
|   |   FPGA Version: 7.0
|   |   
|   |   Time sources:  none, internal, external
|   |   Clock sources: internal, external
|   |   Sensors: ref_locked
|   |     _____________________________________________________
|   |    /
|   |   |       RX DSP: 0
|   |   |   
|   |   |   Freq range: -8.000 to 8.000 MHz
|   |     _____________________________________________________
|   |    /
|   |   |       RX Dboard: A
|   |   |     _____________________________________________________
|   |   |    /
|   |   |   |       RX Frontend: A
|   |   |   |   Name: FE-RX1
|   |   |   |   Antennas: TX/RX, RX2
|   |   |   |   Sensors: temp, rssi, lo_locked
|   |   |   |   Freq range: 50.000 to 6000.000 MHz
|   |   |   |   Gain range PGA: 0.0 to 76.0 step 1.0 dB
|   |   |   |   Bandwidth range: 200000.0 to 56000000.0 step 0.0 Hz
|   |   |   |   Connection Type: IQ
|   |   |   |   Uses LO offset: No
|   |   |     _____________________________________________________
|   |   |    /
|   |   |   |       RX Codec: A
|   |   |   |   Name: B200mini RX dual ADC
|   |   |   |   Gain Elements: None
|   |     _____________________________________________________
|   |    /
|   |   |       TX DSP: 0
|   |   |   
|   |   |   Freq range: -8.000 to 8.000 MHz
|   |     _____________________________________________________
|   |    /
|   |   |       TX Dboard: A
|   |   |     _____________________________________________________
|   |   |    /
|   |   |   |       TX Frontend: A
|   |   |   |   Name: FE-TX1
|   |   |   |   Antennas: TX/RX
|   |   |   |   Sensors: temp, lo_locked
|   |   |   |   Freq range: 50.000 to 6000.000 MHz
|   |   |   |   Gain range PGA: 0.0 to 89.8 step 0.2 dB
|   |   |   |   Bandwidth range: 200000.0 to 56000000.0 step 0.0 Hz
|   |   |   |   Connection Type: IQ
|   |   |   |   Uses LO offset: No
|   |   |     _____________________________________________________
|   |   |    /
|   |   |   |       TX Codec: A
|   |   |   |   Name: B200mini TX dual DAC
|   |   |   |   Gain Elements: None

**Reply by ismagom** [reply]
You are missing an `e6` in the `tx_rate` in the first command

**Reply by rajugbv** [reply]
Hi @ismagom  sorry for the late reply when i replaced tx-rate with 11.52e6 benchmark is working fine but when changed the value of srate to 11.52.e6 from 11.52 i am facing  error while initializing the gnb 

i am using the conf file which was recommended in the below link
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#

please check the below log and reply if you find any error
raju@raju-ThinkPad-P16s-Gen-1:~/srsran/srsRAN_Project/build/apps/gnb$ sudo ./gnb -c /home/raju/srsran/srsRAN_Project/configs/gnb_rf_b210_fdd_srsUE.yml
Lower PHY in dual executor mode.

--== srsRAN gNB (commit 549d12131) ==--

Connecting to AMF on 127.0.1.100:38412
Failed to bind UDP socket to 127.0.0.1:2152. Address already in use
Available radio types: uhd.
[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[INFO] [LOGGING] Fastpath logging disabled at runtime.
[INFO] [B200] Loading firmware image: /usr/local/share/uhd/images/usrp_b200_fw.hex...
No device type given, found device with address 'type=b200,name=B200mini,serial=323E38A,product=B200mini'
Making USRP object with args 'type=b200'
[INFO] [B200] Detected Device: B200mini
[INFO] [B200] Loading FPGA image: /usr/local/share/uhd/images/usrp_b200mini_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test...
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz...
[INFO] [B200] Actually got clock rate 16.000000 MHz.
[INFO] [MULTI_USRP] Setting master clock rate selection to 'manual'.
[INFO] [B200] Asking for clock rate 61.440000 MHz...
[INFO] [B200] Actually got clock rate 61.440000 MHz.
Error: setting Tx sampling rate. Tx Rate 11520000.0 MHz is invalid. The nearest valid value is 61.440000047870704.
srsGNB ERROR: Unable to create radio session.
Aborted

raju@raju-ThinkPad-P16s-Gen-1:~/uhd/host/build/examples$  ./benchmark_rate --rx_rate 11.52e6 --tx_rate 11.52e6

[INFO] [UHD] linux; GNU C++ version 9.4.0; Boost_107100; UHD_4.4.0.0-177-g0e50a37b
[INFO] [B200] Loading firmware image: /usr/local/share/uhd/images/usrp_b200_fw.hex...
[00:00:00.000127] Creating the usrp device with: ...
[INFO] [B200] Detected Device: B210
[INFO] [B200] Loading FPGA image: /usr/local/share/uhd/images/usrp_b210_fpga.bin...
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Detecting internal GPSDO.... 
[INFO] [GPS] No GPSDO found
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Setting master clock rate selection to 'automatic'.
[INFO] [B200] Asking for clock rate 16.000000 MHz... 
[INFO] [B200] Actually got clock rate 16.000000 MHz.
Using Device: Single USRP:
  Device: B-Series Device
  Mboard 0: B210
  RX Channel: 0
    RX DSP: 0
    RX Dboard: A
    RX Subdev: FE-RX2
  RX Channel: 1
    RX DSP: 1
    RX Dboard: A
    RX Subdev: FE-RX1
  TX Channel: 0
    TX DSP: 0
    TX Dboard: A
    TX Subdev: FE-TX2
  TX Channel: 1
    TX DSP: 1
    TX Dboard: A
    TX Subdev: FE-TX1

[00:00:10.956165391] Setting device timestamp to 0...
[INFO] [B200] Asking for clock rate 46.080000 MHz... 
[INFO] [B200] Actually got clock rate 46.080000 MHz.
[00:00:11.355114416] Testing receive rate 11.520000 Msps on 1 channels
Setting TX spp to 2040
[00:00:11.378045787] Testing transmit rate 11.520000 Msps on 1 channels
[00:00:21.383992003] Benchmark complete.


Benchmark rate summary:
  Num received samples:     115467505
  Num dropped samples:      0
  Num overruns detected:    0
  Num transmitted samples:  115266120
  Num sequence errors (Tx): 0
  Num sequence errors (Rx): 0
  Num underruns detected:   0
  Num late commands:        0
  Num timeouts (Tx):        0
  Num timeouts (Rx):        0


Done!

**Comment by pgawlowicz** [comment]
Hi, Please follow the updated tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html

In addition, please run the following script on both PCs:
https://github.com/srsran/srsRAN_Project/blob/main/scripts/srsran_performance
